### PR TITLE
Prepwork for Migrating B3 on Map Reports

### PR DESCRIPTION
**Pull request by biyeun**
@benrudolph @czue @orangejenny 

QAd the one prod-facing change in `reports/javascripts/maps.js` file and runs ok

### CODE REVIEW

File: corehq/apps/reports/static/reports/javascripts/maps.js [hunk 1]
```diff
@@ -124,9 +124,18 @@ function initTable(data, config) {
...
-        aoColumns: colSorting
-    });
+    if (window.USE_BOOTSTRAP_3) {
+        var table = new HQReportDataTables({
+            aoColumns: colSorting,
+            useBootstrap3: true,
+            paginationType: 'bs_normal'
+        });
+    } else {
+        var table = new HQReportDataTables({
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/maps.js]: 'table' is already defined.

**Comment by czue** [comment]
:+1: will let others review. 

can we update the existing refs to the js libraries to use the bower versions now and remove the dependencies from the repo or does that have to happen later?

File: corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            </script>
+
+            <div id="results-capped" class="alert alert-info" style="display: none;">
+                <button type="button" class="close" data-dismiss="alert">&times;</button>
+                There are <strong id="rows-total"></strong> results for the chosen report filters.
+                However, map display has been capped at the first <strong id="rows-capped"></strong> results.
+            </div>
+
+            <div>
+                <div id="map" style="width: 100%;"></div>
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html]: verifying that the `width: 100%` is necessary - why doesn't the div default to full width?

File: corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html [hunk 3]
```diff
@@ -0,0 +1,142 @@
+{% load hq_shared_tags %}
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html]: Does this file need translations, or is it internal-only?

File: corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            <script type="text/template" id="info_popup">
+                <% if (name) { %><h4><%= name %></h4><% } %>
+                <% _.each(info, function(field) { %>
+                <div><%= field.label %></div>
+                <div class="detail_data"><%= field.value %></div>
+                <% }); %>
+            </script>
+
+            <script type="text/template" id="legend_size">
+                <table class="sizelegend">
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html]: nitpick: format class names as word1-word2 (also `detail_data`, `vallabel`, etc.)

Comment (orangejenny) [inline corehq/apps/reports/templates/reports/bootstrap3/partials/base_maps.html]: Nevermind, looks like you're not inventing these names. Not a problem for this PR.

**Comment by orangejenny** [comment]
looks good to me, will give you a chance to respond to cory's question before merging

**Comment by biyeun** [comment]
@czue the bower versions are updated compared to what's in the repo. I can add a legacy version but would rather do that in a separate PR.

@orangejenny yes, `100%` is necessary.

**Comment by biyeun** [comment]
@orangejenny re: inline css. It has it's uses, and I think for heavily js-manipulated components that are really specific to that page, as long as the inline css is short it make sense to keep it, otherwise you get into too many layers of abstraction

**Comment by orangejenny** [comment]
I was wondering why that specific style was necessary, not talking about inline styling in general.

**Comment by biyeun** [comment]
tbh, I as I didn't write the code I can't give you a specific answer. I only did the minimum cleanup and refactor necessary for this. I can investigate further, but seems like a misuse of time if the goal is to stick to bootstrap3 specific changes.

**Comment by orangejenny** [comment]
Yeah, agree that it doesn't sound worth investigating further.

**Comment by biyeun** [comment]
@orangejenny was there something left? can this be merged?

**Comment by orangejenny** [comment]
oops, thought this was one of PRs waiting on qa. Good to merge.

**Pull request merged**